src/satosa/scripts/satosa_saml_metadata.py

```diff
@@ -21,48 +21,58 @@ def _create_split_entity_descriptors(entities, secc, valid):
     output = []
     for module_name, eds in entities.items():
         for i, ed in enumerate(eds):
-            output.append((create_signed_entity_descriptor(ed, secc, valid), "{}_{}.xml".format(module_name, i)))
+            output.append(
+                (create_signed_entity_descriptor(ed, secc, valid),
+                 "{}_{}.xml".format(module_name, i)))
 
     return output
 
 
 def _create_merged_entities_descriptors(entities, secc, valid, name):
     output = []
-    frontend_entity_descriptors = [e for sublist in entities.values() for e in sublist]
-    for frontend in frontend_entity_descriptors:
-        output.append((create_signed_entity_descriptor(frontend, secc, valid), name))
+    entity_descriptors = [e for sublist in entities.values() for e in sublist]
+    for entity_descr in entity_descriptors:
+        output.append(
+            (create_signed_entity_descriptor(entity_descr, secc, valid), name))
 
     return output
 
 
-def create_and_write_saml_metadata(proxy_conf, key, cert, dir, valid, split_frontend_metadata=False,
-                                   split_backend_metadata=False):
-    """
-    Generates SAML metadata for the given PROXY_CONF, signed with the given KEY and associated CERT.
-    """
-    satosa_config = SATOSAConfig(proxy_conf)
-    secc = _get_security_context(key, cert)
-    frontend_entities, backend_entities = create_entity_descriptors(satosa_config)
+def create_saml_metadata(entities, split_option, filename, secc, valid):
+    if split_option:
+        return _create_split_entity_descriptors(entities, secc, valid)
+    else:
+        return _create_merged_entities_descriptors(entities, secc, valid,
+                                                   filename)
 
-    output = []
-    if frontend_entities:
-        if split_frontend_metadata:
-            output.extend(_create_split_entity_descriptors(frontend_entities, secc, valid))
-        else:
-            output.extend(_create_merged_entities_descriptors(frontend_entities, secc, valid, "frontend.xml"))
-    if backend_entities:
-        if split_backend_metadata:
-            output.extend(_create_split_entity_descriptors(backend_entities, secc, valid))
-        else:
-            output.extend(_create_merged_entities_descriptors(backend_entities, secc, valid, "backend.xml"))
 
+def write_saml_metadata(directory, output):
     for metadata, filename in output:
-        path = os.path.join(dir, filename)
+        path = os.path.join(directory, filename)
```

Comment:
`dir` is a builtin function of python and being used here results in shadowing that method. I presumed that this function is only used in terms of the script call (and not destined to be used as part of the library) which has the command argument untouched, that's why I changed it. But sure I'll keep it the way it was for compatibility's sake if you believe it would be used as part of the library

Reply:
`dir` is primarily used when debugging. Since, we have no need for it, and the function has been declared like that long ago, it does not seem to have any worth breaking it.